src/psyclone/tests/dynamo0p3_transformations_test.py

```diff
@@ -2898,19 +3112,43 @@ def test_builtins_usual_then_red_fuse_do():
             schedule, _ = otrans.apply(schedule.children[0], reprod=False)
             schedule, _ = rtrans.apply(schedule.children[0])
             invoke.schedule = schedule
-            code = str(psy.gen)
-            print(code)
-            assert (
-                "      asum = 0.0_r_def\n"
-                "      !\n"
-                "      !$omp parallel default(shared), private(df)\n"
-                "      !$omp do schedule(static), reduction(+:asum)\n"
-                "      DO df=1,undf_any_space_1_f1\n"
-                "        f1_proxy%data(df) = bvalue*f1_proxy%data(df)\n"
-                "        asum = asum+f1_proxy%data(df)\n"
-                "      END DO \n"
-                "      !$omp end do\n"
-                "      !$omp end parallel\n") in code
+            result = str(psy.gen)
+            print(result)
+            if distmem and not annexed:
+                code = (
+                    "      asum = 0.0_r_def\n"
+                    "      !\n"
+                    "      !$omp parallel default(shared), private(df)\n"
+                    "      !$omp do schedule(static), reduction(+:asum)\n"
+                    "      DO df=1,f1_proxy%vspace%get_last_dof_owned()\n"
+                    "        f1_proxy%data(df) = bvalue*f1_proxy%data(df)\n"
+                    "        asum = asum+f1_proxy%data(df)\n"
+                    "      END DO \n"
+                    "      !$omp end do\n"
+                    "      !\n"
+                    "      ! Set halos dirty/clean for fields modified in the "
+                    "above loop\n"
+                    "      !\n"
+                    "      !$omp master\n"
+                    "      CALL f1_proxy%set_dirty()\n"
+                    "      !$omp end master\n"
+                    "      !\n"
+                    "      !$omp end parallel\n"
+                    "      global_sum%value = asum\n"
+                    "      asum = global_sum%get_sum()\n")
+            else:
+                code = (
```

Comment:
Please add comment, e.g. "# distmem is false"

Reply:
Done, thanks.